### DISCUSSION TITLE
Possible to pair a single ps3 controller to multiple blueretro dongles?

### DISCUSSION THREAD

**Discussion by Wild-Dog** [darthcloud/BlueRetro#996] [General]
I'd love to simplify my gaming console setup like that.

**Comment by Wild-Dog** [comment]
By extension, is it possible to change the address of multiple blueretro adapters to match a single controller? I'd only have one console on at a time.

**Comment by konwektor** [comment]
Some time ago somebody asked for similar function : to use  one gamepad, connected to BR adapter, and outpouting data to all adapter ports?, try to use search function for that.
As far I remember guy did it with succes.
Other way is not possible, because single gamepad will connect only to one host/BR ADAPTER. You can`t use single ps3 controller and conect it to multiple ps3 consoles at same time - logic. but You can connect single gamepad, and change BR adapter to sending data from it to all outputs (ports) - effect will be the same.

**Reply by konwektor** [reply]
Small update, I did not read second post :
 If is possible on esp32 to use "own" bluetooth MAC adress,You need to set same mac  on every BR adapter.
Pair gamepad with firts BR module, set everything, and clone spiffs partition (it holds configuration files for BR, and keys from alerady paired  gamepads). 
Gamepad will try to connect/ rejoin to already known device - if evry BR will have keys from this already paired gamepad, and same MAC ADDRESs , this could work.
If transmission between them is protected against spoffing with some sort of extra encryption ,- I think that all will be harder to do, but not impossible.
Half job done  is here :  
[https://github.com/darthcloud/BlueRetro/wiki/Controller-pairing-guide#2---ps3-pairing-guide](https://github.com/darthcloud/BlueRetro/wiki/Controller-pairing-guide#2---ps3-pairing-guide),

**Reply by slonison** [reply]
Hi, I am trying to do this.

I have tried the following steps:

1. Erase_flash on esp1
2. Install HW1 blue retro (boot on 1000, part on 8000, ota on D000, and psx on 10000)
3. Pair Dual Shock 4 Controller
4. read_flash 0x9000 0x6000 nvs.bin
5. read_flash 0xF0000 0x10000 storage.bin

1. Erase flash on esp2
2. Install HW1 blue retro (boot on 1000, part on 8000, ota on D000, and psx on 10000)
3. write_flash 0x9000 nvs.bin (from esp1)
4. read_flash 0xF0000 storage.bin (from esp1)

However, my controller still won't pair with esp2. I'm assuming it's because I'm missing the correct method of clone the mac addresses between the two esp32 devices. Any idea how to do it?

**Comment by darthcloud** [comment]
I think this thread answer this:
https://github.com/darthcloud/BlueRetro/discussions/1221
